schema/filter/strategy/impl/DefaultFilterStrategy.go

```diff
@@ -0,0 +1,63 @@
+package impl
+
+import (
+	mqFilter "github.com/th2-net/th2-common-go/schema/queue/configuration"
+	p_buff "th2-grpc/th2_grpc_common"
+)
+
+type DefaultFilterStrategy struct {
+	extractFields Th2MsgFieldExtraction
+}
+
+func (fs *DefaultFilterStrategy) Construct() {
+	fs.extractFields = Th2MsgFieldExtraction{}
+}
+
+func (fs *DefaultFilterStrategy) Verify(messages *p_buff.MessageGroup, filters []mqFilter.MqRouterFilterConfiguration) bool {
+	// checking
+	for _, filter := range filters {
+		for _, anyMessage := range messages.Messages {
+			if fs.CheckValues(fs.extractFields.GetFields(anyMessage), filter.Metadata.([]mqFilter.FilterFieldsConfiguration)) {
+				return true
+			}
+		}
+	}
+	return false
+
+}
+
+func (fs *DefaultFilterStrategy) CheckValues(messageFields map[string]string, filters []mqFilter.FilterFieldsConfiguration) bool {
```

Comment:
We have two layer of filter:
* list of complex filters
* list of simple filters in each complex filter
A message matches the a pin when it matches to **any** complex filter of the pin
A message matches the complex filter when it matches to **all** simple filters included to complex filter